ffi/android/DemoApp/app/src/main/java/org/lnpbp/demoapp/MainActivity.java

```diff
@@ -0,0 +1,65 @@
+package org.lnpbp.demoapp;
+
+import android.os.Bundle;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.android.material.snackbar.Snackbar;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import android.util.Log;
+import android.view.View;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import org.lnpbp.rgbnode.COpaqueStruct;
+import org.lnpbp.rgbnode.rgb_node;
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        FloatingActionButton fab = findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                System.loadLibrary("rgb_node");
+
+                try {
+                    COpaqueStruct runtime = rgb_node.start_rgb("{\"network\":\"testnet\", \"stash_endpoint\":\"ipc:/home/orlovsky/repo/rgb-node/data/testnet/stashd.rpc\", \"contract_endpoints\":{\"Fungible\":\"ipc:/home/orlovsky/repo/rgb-node/data/testnet/fungibled.rpc\"}}");
+                    rgb_node.issue(runtime, "{\"network\":\"bitcoin\",\"ticker\":\"USDT\",\"name\":\"USD Tether\",\"issue_structure\":\"SingleIssue\",\"allocations\":[{\"coins\":100,\"vout\":0,\"txid\":\"0313ba7cfcaa66029a1a63918ebc426259f00953016c461663315d1bf6b83ab4\"}],\"precision\":0}");
```

Comment:
I think we will be required to provide normal API, not JSON string here... Can we enhance this part somehow?

Reply:
we could do that, but the problem is that we would have to define the same data structures in java and keep them updated with the rust version. since things are still changing pretty quickly i would wait for everything to be more stable before we do that, otherwise we just spend a ton of time keeping them updated and we might also miss something.

many calls are sill missing too, so i would just do that when we get to the point where we have all the calls ready on the rust side and we can map all of them for a proper release.

Reply:
I understand that. However it's really ugly to see all these JSON strings with \". Is there at least normal Java JSON lib that can make creation of these structs less ugly?

BTW, Issue structure is highly unlikely to be changed, so it can be wrapped already into native Java, at least to see/demonstrate how the API will look like